js/interface.js

```diff
@@ -0,0 +1,64 @@
+var appDataSources = [];
+Fliplet.DataSources.get({ appId: Fliplet.Env.get('masterAppId'), attributes: ['id', 'name'] }).then(function (DSs) {
+  appDataSources = DSs.map(el => {
+    return { value: el.id, label: el.name };
+  });
+  Fliplet.Widget.generateInterface({
+    title: 'File list',
+    fields: [
+      {
+        name: 'dataSource',
+        type: 'dropdown',
+        label: 'Data Source',
+        options: appDataSources,
+        default: '',
+        change: function (value) {
+          Fliplet.Helper.field('columnName').toggle(value);
+          if (value) {
+            Fliplet.DataSources.getById(value, {
+              attributes: ['columns']
+            }).then(function (columns) {
+              $(document).find('#columnName').html('')
+              columns.columns.forEach(el => {
+                $(document).find('#columnName').append(`<option value="${el}">${el}</option>`)
```

Comment:
You're setting the HTML to empty and then updating the HTML as many times as there are number of columns. The better option is:

```
const options = arr.map(el => {
  return `some template using ${el}`;
});

$('#dropdown').html(options.join(''));
```

This way you are only updating the HTML once.